src/cfnlint/transform.py

```diff
@@ -0,0 +1,161 @@
+"""
+  Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  software and associated documentation files (the "Software"), to deal in the Software
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import os
+import six
+
+from samtranslator.parser import parser
+from samtranslator.translator.translator import Translator
+from samtranslator.public.exceptions import InvalidDocumentException
+
+import cfnlint.helpers
+
+
+class Transform(object):
+    """
+    Application Serverless Module tranform Wrappor. Based on code from AWS SAM CLI:
+    https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/validate/lib/sam_template_validator.py
+    """
+
+    def __init__(self, filename, template, region):
+        """
+        Initialize Transform class
+        """
+        self._filename = filename
+        self._template = template
+        self._region = region
+
+        self._managed_policy_map = self.load_managed_policies()
+        self._sam_parser = parser.Parser()
+
+
+    def load_managed_policies(self):
+        """
+        Load the ManagedPolicies locally, based on the AWS-CLI:
+        https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/lib/samlib/default_managed_policies.json
+        """
+        return cfnlint.helpers.load_resources('data/Serverless/ManagedPolicies.json')
+
+    def _replace_local_codeuri(self):
```

Comment:
huh thats too bad too.  Be nice if they provided a light weight transformer.  Again maybe we can do to improve their code and lighten ours.

Reply:
This is based on the "validate" step of the CLI, did not really feel the urge on improving their code ATM

Reply:
@kddejong Like a [plugin](https://github.com/awslabs/serverless-application-model/issues/367)!? :) 

I agree through. Using the SAM Translator library is a little cumbersome to setup and use. I ran into this as well when integrating it into SAM CLI but haven't gotten the time to submit an issue on revamping how to use the code yet.

There is defiantly overlap here for us (SAM CLI) and this project to work on some of these things together and get these common actions into one place (cross pollination :) ). Would love to explore some of that as it would be a win for both of us (and others who may need or want the functionality).

For context on why we have this in the SAM CLI:
So the real purpose of this is for SAM CLI to be able to validate the template correctly. The SAM Translator expects a S3 Object (dictionary with bucket, key, version keys) or an S3 URI ([docs](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#properties)) for the CodeUri. This is also true for the `AWS::Serverless::Api`'s `DefinitionUri` Property. In SAM CLI however, we want customers to use this CodeUri or DefinitionUri` to point to their code on their local machine. If we don't replace, then the validation will fail. The `package` command already does the replacing of these keys before a customer deploys their code and template, so this is not to break the mental model for the tool.